Core/Utilities/TimeLeft/LSFTimeLeft.py

```diff
@@ -110,14 +102,15 @@ def __init__( self ):
                   self.normRef = norm
                   self.log.info( 'Reference Normalization taken from Host Model',
                                  '%s: %s' % ( self.cpuRef, self.normRef ) )
-              except:
-                pass
+              except Exception as e:
+                self.log.exception( 'Exception parsing lsfinfo output', '', e )
 
       if not self.normRef:
         # Now parse LSF configuration files
         if not os.path.isfile( './lsf.sh' ):
           os.symlink( os.path.join( os.environ['LSF_ENVDIR'], 'lsf.conf' ) , './lsf.sh' )
-        ret = sourceEnv( 10, ['./lsf' ] )
+        # As the variables are not exported, we must force it
+        ret = sourceEnv( 10, ['./lsf', '&& export LSF_CONFDIR' ] )
```

Comment:
The method sourceEnv() explicitly uses bash, therefore it is safe, even if the main shell is another one.. This code is rather tricky!